### ISSUE TITLE
[NixOS][Hyprland] Random Crashes, Signal 11

### ISSUE THREAD

**Issue by Hannes1909** [xournalpp/xournalpp#5761]
### Operating System

Linux

### (Linux only) Distribution

NixOS unstable

### (Linux only) Desktop Environment

Hyprland

### (Linux Only) Display Server

Wayland

### Installation Method

NixPkg, via Home-Manager

### Xournal++ Version

Xournal++ version 1.2.3

### libgtk Version

Gtk version 3.24.42

### Bug Description

Random crashes when annotating PDFs or drawing. Im not quite sure if the binutils-wrapper error is related, since Xournal++ already crashed when the lines are logged in stdout. When starting Xournal++ again the file gets recovered pretty much lossless from what I can tell (maybe some pixels of the last drawn line, but nothing that cannot be overlooked).

My home-manager file containing the Xournal++ package is pretty straightforward:
```
{ config, pkgs, ... }:

{
  home.packages = [
    pkgs.xournalpp
  ];
}
```

I also tested Xournal with a similar config, which does not seem to crash, but of course has much less features and quality of life addons.
I already disabled Sound Recording in X++'s settings, since I dont need it and I got some errors in stdout referring to ALSA like unknown card, unable to find definition etc. - those errors are gone since the SoundRecording is disabled.

If it is relevant to the troubleshooting, I use an XP Pen 12 graphics tablet with OpenTabletDriver. The Stylus and tablet seem to work fine in X++, pressure, eraser, buttons, everything ok.

Thanks for any hints - I can provide any log, trace, debug-output that might be relevant if you point me in the right direction :)

Console Output:
```
xopp-Message: 20:03:40.818: TEXTDOMAINDIR = (null), Platform-specific locale dir = /nix/store/d8r30gan1nilzjqdffmm72m70d9qh9rn-xournalpp-1.2.3/share/xournalpp/../locale, chosen directory = /nix/store/d8r30gan1nilzjqdffmm72m70d9qh9rn-xournalpp-1.2.3/share/xournalpp/../locale
xopp-Message: 20:03:40.820: Plugin "MigrateFontSizes" UI initialized

(com.github.xournalpp.xournalpp:12361): xopp-WARNING **: 20:04:27.225: [Crash Handler] Crashed with signal 11

(com.github.xournalpp.xournalpp:12361): xopp-WARNING **: 20:04:27.225: [Crash Handler] Wrote crash log to: /home/hannes/.cache/xournalpp/errorlogs/errorlog.20240716-200427.log
/nix/store/9rz99vibv1782dxjkwvxr29s7f2f86wl-binutils-wrapper-2.42/bin/addr2line: 'xournalpp': No such file
/nix/store/9rz99vibv1782dxjkwvxr29s7f2f86wl-binutils-wrapper-2.42/bin/addr2line: 'xournalpp': No such file
/nix/store/9rz99vibv1782dxjkwvxr29s7f2f86wl-binutils-wrapper-2.42/bin/addr2line: 'xournalpp': No such file
/nix/store/9rz99vibv1782dxjkwvxr29s7f2f86wl-binutils-wrapper-2.42/bin/addr2line: 'xournalpp': No such file
/nix/store/9rz99vibv1782dxjkwvxr29s7f2f86wl-binutils-wrapper-2.42/bin/addr2line: 'xournalpp': No such file
/nix/store/9rz99vibv1782dxjkwvxr29s7f2f86wl-binutils-wrapper-2.42/bin/addr2line: 'xournalpp': No such file
/nix/store/9rz99vibv1782dxjkwvxr29s7f2f86wl-binutils-wrapper-2.42/bin/addr2line: 'xournalpp': No such file
/nix/store/9rz99vibv1782dxjkwvxr29s7f2f86wl-binutils-wrapper-2.42/bin/addr2line: 'xournalpp': No such file

(com.github.xournalpp.xournalpp:12361): xopp-WARNING **: 20:04:27.331: Versuche eine Notfall-Speicherung des aktuell geöffneten Dokuments…

(com.github.xournalpp.xournalpp:12361): xopp-WARNING **: 20:04:27.343: Dokument erfolgreich nach "/home/hannes/.config/xournalpp/emergencysave.xopp" gespeichert.
```

Latest Errorlog:
```
Date: Tue Jul 16 20:04:27 2024
Error: signal 11
Xournal++ version 1.2.3
Gtk version 3.24.42

[bt]: (0) xournalpp() [0x4d091f]
[bt]: (1) /nix/store/m71p7f0nymb19yn1dascklyya2i96jfw-glibc-2.39-52/lib/libc.so.6(+0x405c0) [0x7fb33fd635c0]
[bt]: (2) /nix/store/n8vgi779q26fqzw88a3yvcjkw9lp79rp-wayland-1.22.0/lib/libwayland-client.so.0(wl_proxy_marshal_flags+0xa9) [0x7fb33f1d8409]
[bt]: (3) /nix/store/3mzh8lvgbynm9daj7c82k2sfsfhrsfsy-gtk+3-3.24.42/lib/libgdk-3.so.0(+0x6f230) [0x7fb340b5b230]
[bt]: (4) /nix/store/3mzh8lvgbynm9daj7c82k2sfsfhrsfsy-gtk+3-3.24.42/lib/libgdk-3.so.0(gdk_window_set_cursor+0xf2) [0x7fb340b3f2b2]
[bt]: (5) xournalpp(_ZN15XournalppCursor9setCursorEi+0xa6) [0x51e0c6]
[bt]: (6) xournalpp(_ZN15XournalppCursor12updateCursorEv+0x303) [0x51e903]
[bt]: (7) xournalpp(_ZN17MouseInputHandler10handleImplERK10InputEvent+0x131) [0x5213c1]
[bt]: (8) xournalpp(_ZN12InputContext6handleEP9_GdkEvent+0x34c) [0x52024c]
[bt]: (9) xournalpp(_ZN3xoj4util6detail9wrap_implIXadL_ZN12InputContext13eventCallbackEP10_GtkWidgetP9_GdkEventPS3_EEbS8_JS5_S7_EEENSt11conditionalIX9is_same_vIbT0_EEiSA_E4typeEDpT2_Pv+0x9) [0x520a59]
[bt]: (10) /nix/store/3mzh8lvgbynm9daj7c82k2sfsfhrsfsy-gtk+3-3.24.42/lib/libgtk-3.so.0(+0x9e104) [0x7fb340c9e104]
[bt]: (11) /nix/store/lw6pxp6px4s8cz85y8jbcnfciy6xdmay-glib-2.80.3/lib/libgobject-2.0.so.0(+0x17f41) [0x7fb340769f41]
[bt]: (12) /nix/store/lw6pxp6px4s8cz85y8jbcnfciy6xdmay-glib-2.80.3/lib/libgobject-2.0.so.0(+0x2dd98) [0x7fb34077fd98]
[bt]: (13) /nix/store/lw6pxp6px4s8cz85y8jbcnfciy6xdmay-glib-2.80.3/lib/libgobject-2.0.so.0(g_signal_emit_valist+0x32) [0x7fb3407867f2]
[bt]: (14) /nix/store/lw6pxp6px4s8cz85y8jbcnfciy6xdmay-glib-2.80.3/lib/libgobject-2.0.so.0(g_signal_emit+0x8f) [0x7fb3407868af]
[bt]: (15) /nix/store/3mzh8lvgbynm9daj7c82k2sfsfhrsfsy-gtk+3-3.24.42/lib/libgtk-3.so.0(+0x36b7d6) [0x7fb340f6b7d6]
[bt]: (16) /nix/store/3mzh8lvgbynm9daj7c82k2sfsfhrsfsy-gtk+3-3.24.42/lib/libgtk-3.so.0(+0x2136d6) [0x7fb340e136d6]
[bt]: (17) /nix/store/3mzh8lvgbynm9daj7c82k2sfsfhrsfsy-gtk+3-3.24.42/lib/libgtk-3.so.0(gtk_main_do_event+0x6be) [0x7fb340e1534e]
[bt]: (18) /nix/store/3mzh8lvgbynm9daj7c82k2sfsfhrsfsy-gtk+3-3.24.42/lib/libgdk-3.so.0(+0x405e5) [0x7fb340b2c5e5]
[bt]: (19) /nix/store/3mzh8lvgbynm9daj7c82k2sfsfhrsfsy-gtk+3-3.24.42/lib/libgdk-3.so.0(+0x77f42) [0x7fb340b63f42]
[bt]: (20) /nix/store/lw6pxp6px4s8cz85y8jbcnfciy6xdmay-glib-2.80.3/lib/libglib-2.0.so.0(+0x5e074) [0x7fb340660074]
[bt]: (21) /nix/store/lw6pxp6px4s8cz85y8jbcnfciy6xdmay-glib-2.80.3/lib/libglib-2.0.so.0(+0x61297) [0x7fb340663297]
[bt]: (22) /nix/store/lw6pxp6px4s8cz85y8jbcnfciy6xdmay-glib-2.80.3/lib/libglib-2.0.so.0(g_main_context_iteration+0x2c) [0x7fb3406638bc]
[bt]: (23) /nix/store/lw6pxp6px4s8cz85y8jbcnfciy6xdmay-glib-2.80.3/lib/libgio-2.0.so.0(g_application_run+0x16d) [0x7fb3408aa5cd]
[bt]: (24) xournalpp(_ZN11XournalMain3runEiPPc+0x636) [0x4d5946]
[bt]: (25) /nix/store/m71p7f0nymb19yn1dascklyya2i96jfw-glibc-2.39-52/lib/libc.so.6(+0x2a14e) [0x7fb33fd4d14e]
[bt]: (26) /nix/store/m71p7f0nymb19yn1dascklyya2i96jfw-glibc-2.39-52/lib/libc.so.6(__libc_start_main+0x89) [0x7fb33fd4d209]
[bt]: (27) xournalpp(_start+0x25) [0x4cf795]

Try to get a better stacktrace...
[bt] #0 xournalpp() [0x4d1008]
[bt] #1 /nix/store/m71p7f0nymb19yn1dascklyya2i96jfw-glibc-2.39-52/lib/libc.so.6(+0x405c0) [0x7fb33fd635c0]
__restore_rt at ??:?
[bt] #2 /nix/store/n8vgi779q26fqzw88a3yvcjkw9lp79rp-wayland-1.22.0/lib/libwayland-client.so.0(wl_proxy_marshal_flags+0xa9) [0x7fb33f1d8409]
wl_proxy_marshal_flags at ??:?
[bt] #3 /nix/store/3mzh8lvgbynm9daj7c82k2sfsfhrsfsy-gtk+3-3.24.42/lib/libgdk-3.so.0(+0x6f230) [0x7fb340b5b230]
gdk_wayland_device_update_window_cursor at ??:?
[bt] #4 /nix/store/3mzh8lvgbynm9daj7c82k2sfsfhrsfsy-gtk+3-3.24.42/lib/libgdk-3.so.0(gdk_window_set_cursor+0xf2) [0x7fb340b3f2b2]
gdk_window_set_cursor at ??:?
[bt] #5 xournalpp(_ZN15XournalppCursor9setCursorEi+0xa6) [0x51e0c6]
[bt] #6 xournalpp(_ZN15XournalppCursor12updateCursorEv+0x303) [0x51e903]
[bt] #7 xournalpp(_ZN17MouseInputHandler10handleImplERK10InputEvent+0x131) [0x5213c1]
[bt] #8 xournalpp(_ZN12InputContext6handleEP9_GdkEvent+0x34c) [0x52024c]
[bt] #9 xournalpp(_ZN3xoj4util6detail9wrap_implIXadL_ZN12InputContext13eventCallbackEP10_GtkWidgetP9_GdkEventPS3_EEbS8_JS5_S7_EEENSt11conditionalIX9is_same_vIbT0_EEiSA_E4typeEDpT2_Pv+0x9) [0x520a59]
[bt] #10 /nix/store/3mzh8lvgbynm9daj7c82k2sfsfhrsfsy-gtk+3-3.24.42/lib/libgtk-3.so.0(+0x9e104) [0x7fb340c9e104]
_gtk_marshal_BOOLEAN__BOXEDv at ??:?
[bt] #11 /nix/store/lw6pxp6px4s8cz85y8jbcnfciy6xdmay-glib-2.80.3/lib/libgobject-2.0.so.0(+0x17f41) [0x7fb340769f41]
_g_closure_invoke_va at ??:?
[bt] #12 /nix/store/lw6pxp6px4s8cz85y8jbcnfciy6xdmay-glib-2.80.3/lib/libgobject-2.0.so.0(+0x2dd98) [0x7fb34077fd98]
signal_emit_valist_unlocked at ??:?
[bt] #13 /nix/store/lw6pxp6px4s8cz85y8jbcnfciy6xdmay-glib-2.80.3/lib/libgobject-2.0.so.0(g_signal_emit_valist+0x32) [0x7fb3407867f2]
g_signal_emit_valist at ??:?
[bt] #14 /nix/store/lw6pxp6px4s8cz85y8jbcnfciy6xdmay-glib-2.80.3/lib/libgobject-2.0.so.0(g_signal_emit+0x8f) [0x7fb3407868af]
g_signal_emit at ??:?
[bt] #15 /nix/store/3mzh8lvgbynm9daj7c82k2sfsfhrsfsy-gtk+3-3.24.42/lib/libgtk-3.so.0(+0x36b7d6) [0x7fb340f6b7d6]
gtk_widget_event_internal.part.0 at ??:?
[bt] #16 /nix/store/3mzh8lvgbynm9daj7c82k2sfsfhrsfsy-gtk+3-3.24.42/lib/libgtk-3.so.0(+0x2136d6) [0x7fb340e136d6]
propagate_event at ??:?
[bt] #17 /nix/store/3mzh8lvgbynm9daj7c82k2sfsfhrsfsy-gtk+3-3.24.42/lib/libgtk-3.so.0(gtk_main_do_event+0x6be) [0x7fb340e1534e]
gtk_main_do_event at ??:?
[bt] #18 /nix/store/3mzh8lvgbynm9daj7c82k2sfsfhrsfsy-gtk+3-3.24.42/lib/libgdk-3.so.0(+0x405e5) [0x7fb340b2c5e5]
_gdk_event_emit at ??:?
[bt] #19 /nix/store/3mzh8lvgbynm9daj7c82k2sfsfhrsfsy-gtk+3-3.24.42/lib/libgdk-3.so.0(+0x77f42) [0x7fb340b63f42]
gdk_event_source_dispatch at ??:?
[bt] #20 /nix/store/lw6pxp6px4s8cz85y8jbcnfciy6xdmay-glib-2.80.3/lib/libglib-2.0.so.0(+0x5e074) [0x7fb340660074]
g_main_dispatch at ??:?
[bt] #21 /nix/store/lw6pxp6px4s8cz85y8jbcnfciy6xdmay-glib-2.80.3/lib/libglib-2.0.so.0(+0x61297) [0x7fb340663297]
g_main_context_iterate_unlocked.isra.0 at ??:?
[bt] #22 /nix/store/lw6pxp6px4s8cz85y8jbcnfciy6xdmay-glib-2.80.3/lib/libglib-2.0.so.0(g_main_context_iteration+0x2c) [0x7fb3406638bc]
g_main_context_iteration at ??:?
[bt] #23 /nix/store/lw6pxp6px4s8cz85y8jbcnfciy6xdmay-glib-2.80.3/lib/libgio-2.0.so.0(g_application_run+0x16d) [0x7fb3408aa5cd]
g_application_run at ??:?
[bt] #24 xournalpp(_ZN11XournalMain3runEiPPc+0x636) [0x4d5946]
[bt] #25 /nix/store/m71p7f0nymb19yn1dascklyya2i96jfw-glibc-2.39-52/lib/libc.so.6(+0x2a14e) [0x7fb33fd4d14e]
__libc_start_call_main at ??:?
[bt] #26 /nix/store/m71p7f0nymb19yn1dascklyya2i96jfw-glibc-2.39-52/lib/libc.so.6(__libc_start_main+0x89) [0x7fb33fd4d209]
__libc_start_main_alias_2 at ??:?
[bt] #27 xournalpp(_start+0x25) [0x4cf795]

Execution log:

DEBUG   : GLib-GIO :: Using cross-namespace EXTERNAL authentication (this will deadlock if server is GDBus < 2.73.3)
DEBUG   : GLib-GIO :: _g_io_module_get_default: Found default implementation dconf (DConfSettingsBackend) for ‘gsettings-backend’
DEBUG   : GLib-GIO :: _g_io_module_get_default: Found default implementation gvfs (GDaemonVfs) for ‘gio-vfs’
DEBUG   : GLib :: unsetenv() is not thread-safe and should not be used after threads are created
MESSAGE : xopp :: TEXTDOMAINDIR = (null), Platform-specific locale dir = /nix/store/d8r30gan1nilzjqdffmm72m70d9qh9rn-xournalpp-1.2.3/share/xournalpp/../locale, chosen directory = /nix/store/d8r30gan1nilzjqdffmm72m70d9qh9rn-xournalpp-1.2.3/share/xournalpp/../locale
INFO    : xopp :: Loading plugins from: /nix/store/d8r30gan1nilzjqdffmm72m70d9qh9rn-xournalpp-1.2.3/share/xournalpp/plugins
INFO    : xopp :: Loading plugins from: /home/hannes/.config/xournalpp/plugins
MESSAGE : xopp :: Plugin "MigrateFontSizes" UI initialized
DEBUG   : GdkPixbuf :: gdk_pixbuf_from_pixdata() called on:
DEBUG   : GdkPixbuf :: 	Encoding raw
DEBUG   : GdkPixbuf :: 	Dimensions: 20 x 25
DEBUG   : GdkPixbuf :: 	Rowstride: 80, Length: 2024
DEBUG   : GdkPixbuf :: 	Copy pixels == false
DEBUG   : GdkPixbuf :: gdk_pixbuf_from_pixdata() called on:
DEBUG   : GdkPixbuf :: 	Encoding raw
DEBUG   : GdkPixbuf :: 	Dimensions: 20 x 25
DEBUG   : GdkPixbuf :: 	Rowstride: 80, Length: 2024
DEBUG   : GdkPixbuf :: 	Copy pixels == false
WARNING : xopp :: [Crash Handler] Crashed with signal 11
WARNING : xopp :: [Crash Handler] Wrote crash log to: /home/hannes/.cache/xournalpp/errorlogs/errorlog.20240716-200427.log
```

### Expected Behaviour

No (seemingly) random crashes - mostly I can continue to work right where X++ crashed, but it kills the flow needing to restart the app.

### Steps to Reproduce

- Open X++
- Annotate PDF for some time
- Crash (Signal 11)

Maybe its not reproducable on other systems, but i could not find any more correlated steps that lead to the crash up until now, sorry :(

### Additional Context

```
xinput list
WARNING: running xinput against an Xwayland server. See the xinput man page for details.
⎡ Virtual core pointer                    	id=2	[master pointer  (3)]
⎜   ↳ Virtual core XTEST pointer              	id=4	[slave  pointer  (2)]
⎜   ↳ xwayland-pointer:1                      	id=6	[slave  pointer  (2)]
⎜   ↳ xwayland-relative-pointer:1             	id=7	[slave  pointer  (2)]
⎜   ↳ xwayland-pointer-gestures:1             	id=8	[slave  pointer  (2)]
⎜   ↳ xwayland-tablet stylus:1                	id=10	[slave  pointer  (2)]
⎜   ↳ xwayland-tablet eraser:1                	id=11	[slave  pointer  (2)]
⎜   ↳ xwayland-tablet cursor:1                	id=12	[slave  pointer  (2)]
⎣ Virtual core keyboard                   	id=3	[master keyboard (2)]
    ↳ Virtual core XTEST keyboard             	id=5	[slave  keyboard (3)]
    ↳ xwayland-keyboard:1                     	id=9	[slave  keyboard (3)]
```

**Comment by Hannes1909** [comment]
just did a strace, in case it helps, right befor the crash:

```
--- SIGSEGV {si_signo=SIGSEGV, si_code=SEGV_MAPERR, si_addr=0x10} ---
ioctl(2, TCGETS, {c_iflag=ICRNL|IUTF8, c_oflag=NL0|CR0|TAB0|BS0|VT0|FF0|OPOST|ONLCR, c_cflag=B38400|CS8|CREAD, c_lflag=ISIG|ICANON|ECHO|ECHOE|ECHOK|IEXTEN|ECHOCTL|ECHOKE, ...}) = 0
getpid()                                = 18040
write(2, "\n(com.github.xournalpp.xournalpp"..., 128
(com.github.xournalpp.xournalpp:18040): xopp-WARNING **: 20:33:45.082: [Crash Handler] Crashed with signal ) = 128
write(2, "11\n", 311
)                     = 3
```

**Comment by Hannes1909** [comment]
Oh thanks - ill track it upstream on Hyprland, couldnt find that issue untilk now - thank you for pointing me there!

**Comment by Hannes1909** [comment]
Just for Info - fixed with hyprland 0.42 :)
